src/unitxt/task.py

```diff
@@ -10,42 +13,145 @@ class Tasker:
+
         for augmentable_input in self.augmentable_inputs:
             assert (
                 augmentable_input in self.inputs
             ), f"augmentable_input f{augmentable_input} is not part of {self.inputs}"
 
-    def process(
-        self, instance: Dict[str, Any], stream_name: Optional[str] = None
-    ) -> Dict[str, Any]:
+    def check_metrics_type(self) -> None:
+        prediction_type = self.get_prediction_type(self.prediction_type)
+        for metric_name in self.metrics:
+            metric = fetch_artifact(metric_name)[0]
+            metric_prediction_type = metric.get_prediction_type()
+
+            if (
+                prediction_type == metric_prediction_type
+                or prediction_type == Any
+                or metric_prediction_type == Any
+            ):
+                continue
+
+            raise ValueError(
+                f"The task's prediction type ({prediction_type}) and '{metric_name}' "
+                f"metric's prediction type ({metric_prediction_type}) are different."
+            )
+
+    def get_input_value(self, instance: Dict[str, Any], key: str) -> Any:
         try:
-            inputs = {key: instance[key] for key in self.inputs}
+            return instance[key]
         except KeyError as e:
             raise KeyError(
-                f"Unexpected FormTask input column names ({[key for key in self.inputs if key not in instance]})."
-                f"The available input names: {list(instance.keys())}"
+                f"Unexpected Task 'inputs' names: {[k for k in self.inputs if k not in instance]}."
+                f"The available names: {list(instance.keys())}."
             ) from e
+
+    def get_output_value(self, instance: Dict[str, Any], key: str) -> Any:
         try:
-            outputs = {key: instance[key] for key in self.outputs}
+            return instance[key]
         except KeyError as e:
             raise KeyError(
-                f"Unexpected FormTask output column names: {[key for key in self.outputs if key not in instance]}"
-                f" \n available names:{list(instance.keys())}\n given output names:{self.outputs}"
+                f"Unexpected Task 'outputs' names: {[k for k in self.outputs if k not in instance]}."
+                f"The available names: {list(instance.keys())}."
             ) from e
 
+    def check_instance_value_type(
+        self,
+        value: Any,
+        data_type: str,
+        field_name: str,
+        io_type: Literal["inputs", "outputs"],
+    ) -> Any:
+        data_type = self.get_prediction_type(data_type)
+
+        if isoftype(value, data_type):
+            return value
+
+        raise ValueError(
+            f"Passed {io_type} value '{value}' of field {field_name} is not of required "
+            f"type ({data_type})."
+        )
+
+    def process(
+        self, instance: Dict[str, Any], stream_name: Optional[str] = None
+    ) -> Dict[str, Any]:
+        self.check_metrics_type()
```

Comment:
```suggestion
```
This should be execcuted in the verify() function rather than for every instance.